### DISCUSSION TITLE
High Client Count Instability Issues

### DISCUSSION THREAD

**Discussion by Noelzu** [Ylianst/MeshCentral#6113] [Issues] [answered]
We recently experienced an increase in server instability issues.
Can a high number of clients cause such issues? We currently have about 300 clients.
Any advice would be appreciated.

**Comment by si458** [comment]
you havent really told us your server spec, where it hosted, multi-domain(multi-tenant), session recording enabled/disabled?
we need more info im afraid

**Comment by silversword411** [comment]
10000+ agents isn't a problem with a properly spec'd system

**Comment by PathfinderNetworks** [comment]
I have over 550 devices with Mesh running on a Windows Server 2022 Hyper-V machine that only has 4GB of RAM allocated to it.  Absolutely no performance issues whatsoever.  Heck, even with just 4GB of RAM allocated to the VM it still shows as having 2.1GB available.  The server is running nothing else at all though.

**Comment by xardius** [comment, marked as answer]
The problem was a wrongly configured reverse proxy. The values ​​for "MaxRequestWorkers" and "ServerLimit" in the mpm_event.conf were too low. Thanks for the help offered.
